### PR TITLE
envoy crd / ingress controller support

### PR DESCRIPTION
**Pull request by michi-covalent**
3 commits:

- add ciliumegressnatpolicy and ciliumenvoyconfig to cluster role.
- update cluster role for ingress controller
- create cilium ingress class.

### CODE REVIEW

**Review by jrajahalme — APPROVED** [review]

File: internal/k8s/helpers.go [hunk 1]
```diff
@@ -41,6 +40,17 @@ func NewClusterRoleBinding(name, namespace, serviceAccount string) *rbacv1.Clust
...
+func NewIngressClass(name string) *v1.IngressClass {
+	return &v1.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: v1.IngressClassSpec{
+			Controller: defaults.IngressControllerName,
+		},
+	}
+}
```
Comment (tklauser) [inline internal/k8s/helpers.go:52]: Let's change the ingress controller name to be passed as an argument, akin to other functions in this package:

```suggestion
func NewIngressClass(name, controllerName string) *v1.IngressClass {
	return &v1.IngressClass{
		ObjectMeta: metav1.ObjectMeta{
			Name: name,
		},
		Spec: v1.IngressClassSpec{
			Controller: controllerName,
		},
	}
}
```

File: internal/k8s/helpers.go [hunk 2]
```diff
@@ -4,13 +4,12 @@
 package k8s
 
 import (
-	//appsv1 "k8s.io/api/apps/v1"
+	"github.com/cilium/cilium-cli/defaults"
+
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/networking/v1"
```
Comment (tklauser) [inline internal/k8s/helpers.go:10]:
```suggestion
	networkingv1 "k8s.io/api/networking/v1"
```

**Review by tklauser — CHANGES_REQUESTED** [review]
Two small nits, otherwise LGTM.

**Review by tklauser — APPROVED** [review]

**Comment by michi-covalent** [comment]
aks failed to pull quay.io/cilium/cilium-cli-ci:313536d43e475cac9bcabd155e26f67d2d3dfd20. i can pull it now, so i think it's a transient failure. restarting.

**Review by twpayne — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Comment by michi-covalent** [comment]
https://github.com/cilium/cilium-cli/runs/4451759861?check_suite_focus=true

failed to download eksctl. restarting!

**Pull request merged**